src/sentry/api/validators/sentry_apps/schema.py

```diff
@@ -237,6 +241,32 @@ def check_each_element_for_error(instance, element_types=None):
             )
 
 
+def validate_text_component_defaults(element, found_type):
+    optional_fields = (
+        element["settings"].get("optional_fields")
+        if found_type == "alert-rule-action"
+        else element.get("optional_fields")
+    )
+    required_fields = (
+        element["settings"].get("required_fields")
+        if found_type == "alert-rule-action"
+        else element.get("required_fields")
+    )
+    all_fields = (
```

Comment:
I think you combine the two ternaries by putting `data` in a variable. Also if you put `[]` as a default for your `get()` calls you can eliminate the `all_fields` condition.
```py
def validate_text_component_defaults(element: Mapping[str, Any], found_type: str) -> None:
    data = element["settings"] if found_type == "alert-rule-action" else element
    optional_fields = data.get("optional_fields", [])
    required_fields = data.get("required_fields", [])

    for field in optional_fields + required_fields:
        if field.get("type") in TEXT_COMPONENTS:
            default = field.get("default")
            if default and default not in DEFAULT_TEXT_TYPES:
                raise SchemaValidationError(
                    f"Elements of type {TEXT_COMPONENTS} may only have a default value"
                    f" of the following: {DEFAULT_TEXT_TYPES}, but {default} was found."
                )
```

Reply:
Nice, I figured there was a cleaner way to do this.